akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -75,8 +76,22 @@ object Sink {
 
   /**
    * A `Sink` that materializes into a `Future` of the first value received.
+   * If the stream completes before signaling at least a single element, the Future will be failed with an [[NoSuchElementException]].
+   * If the stream signals an error errors before signaling at least a single element, the Future will be failed with the streams exception.
+   *
+   * See also [[headOption]].
+   */
+  def head[T]: Sink[T, Future[T]] = new Sink[T, Future[Option[T]]](new HeadOptionSink[T](DefaultAttributes.headSink, shape("HeadSink")))
```

Comment:
You mean like this?

```scala
Sink.headOption[T]
      .mapMaterializedValue(e ⇒ e.map(_.getOrElse(throw new NoSuchElementException("head of empty stream")))(ExecutionContexts.sameThreadExecutionContext))
      .withAttributes(DefaultAttributes.headSink)
```

Won't the shape then be `shape("HeadOptionSink")`, and it could appear in logs as `HeadOptionSink.in`? Could be slightly confusing "hey, but I'm using HeadSink!" Or we don't mind that?